pkg/auth/auth.go

```diff
@@ -0,0 +1,54 @@
+package auth
+
+import (
+	"fmt"
+
+	envoy_config_core_v3 "github.com/envoyproxy/go-control-plane/envoy/config/core/v3"
+	xds_ext_authz "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/http/ext_authz/v3"
+	xds_hcm "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/network/http_connection_manager/v3"
+	"github.com/envoyproxy/go-control-plane/pkg/wellknown"
+	"github.com/golang/protobuf/ptypes"
+
+	"github.com/openservicemesh/osm/pkg/logger"
+)
+
+var (
+	log = logger.New("auth")
+)
+
+// GetExtAuthzHTTPFilter returns an envoy HttpFilter given an ExternAuthConfig configuration
+func GetExtAuthzHTTPFilter(extAuthConfig ExternAuthConfig) *xds_hcm.HttpFilter {
```

Comment:
Sorry this is not what I meant when I suggested moving the struct type to pkg/auth.

This function should very much be under `pkg/envoy/lds`

Reply:
I wonderd about that. moving back.